### PR TITLE
Add TCP, boards firmwares, refactoring and more...

### PR DESCRIPTION
**Pull request by Vourhey**
## Refactoring

Now there are 3 folders:
* drivers - should contain drivers for sensors (only SDS011 is present at the moment)
* feeders - outputs in terms of this package
* stations - inputs/sources of data

There are `IFeeder` and `IStation` interfaces. Their meaning is to provide a template to build new inputs/outputs. Have a look at the corresponding files

## New features

It's now possible to specify `publish interval` less than `work period`. It means it's not required to wait for a new record of data. If there is no new data a station provides the last measurement.

Datalog feeder now collects measurements for a period of time and dumps the whole history at once.

TCP Station (#3) is present. It creates a server on a provided port and waits for messages like:
```
b'message\n'b'message\n'
```
where every `message` ends with a newline character and is valid JSON string with measurements.

Both COMStation and TCPStation create a nonblocking threads for data input

## Added boards

* Arduino firmware
* ESP8266 firmware - testing is needed

### CODE REVIEW

File: src/stations/tcpstation.py [hunk 1]
```diff
@@ -35,42 +40,108 @@ def _get_non_blocking_server_socket(self):
...
-
-                    if resource not in self.OUTPUTS:
-                        self.OUTPUTS.append(resource)
+
+                print("Peer name: " + str(resource.getpeername()))
+
+                peer = resource.getpeername()
+                if peer not in self.session:
+                    print("Unknown peer yet")
+                    data = resource.recv(36)
```
Comment (akru) [inline src/stations/tcpstation.py:91]: Magic number

File: src/drivers/payload.proto [hunk 2]
```diff
@@ -0,0 +1,15 @@
+syntax = "proto2";
+
+package arduino_sds;
+
+message Header {
+  required bytes public_key = 1;
```
Comment (akru) [inline src/drivers/payload.proto:6]: 32 byte static size array is enough

File: boards/esp/ESP_TCP/ESP_TCP.ino [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    memcpy(buffer+8, (byte*) &GEO_LAT, 4);
+    memcpy(buffer+12, (byte*) &GEO_LON, 4);
+
+    Ed25519::sign(signature, privateKey, publicKey, buffer, 16);
+    if (client.connected()) {
+      client.write(buffer, sizeof(buffer));
+      client.write(signature, sizeof(signature));
+      client.flush();
+    } else {
+      Serial.println("Not connected");
```
Comment (akru) [inline boards/esp/ESP_TCP/ESP_TCP.ino:90]: Better try to reconnect instead of just screaming about it and do nothing

Comment (Vourhey) [reply boards/esp/ESP_TCP/ESP_TCP.ino:90]: According to the documentation https://arduino-esp8266.readthedocs.io/en/latest/esp8266wifi/readme.html ESP reconnects automatically

**Comment by Vourhey** [comment]
I'm going to merge PR and let others to test new features. After 1-2 more participants I will release connectivity v0.2

Issues #3 and #6 are implemented

**Pull request merged**